### PR TITLE
tickets/DM-3688 split linux/OSX expected results

### PR DESCRIPTION
**Pull request by jhoblitt**

### CODE REVIEW

**Comment by jhoblitt** [comment]
I seem to have forgotten some of the past discussion on this topic, remind me why we are retaining both the numpy and the numdiff comparison methods?

**Comment by jdswinbank** [comment]
Russell asked the same thing. My answer to him was:

>> Why support both numdiff and numpy? It would be useful to document the reason for that. If numpy suffices it would be nice to stick with it, since it requires no extra dependencies.

> I had two motivations here:
* While developing, I wanted to run both in parallel to demonstrate I was getting the same numbers;
* I wanted to be as minimally invasive as possible on the CI system, since I have no way to test the impact of my changes there. In particular, CI jumps through hoops to build and install numdiff for this purpose, and I didn't want to start trying to unpick that.
>
> I'm happy to drop the numdiff support if both you & Joshua Hoblitt reckon it's unnecessary overhead. Do note, though, that it's not really a "dependency", in that the script will run fine without it unless you specifically request that numdiff be used to perform the comparison.

Given that you're asking the same thing, maybe we should disable numdiff. I've no problem with pulling it from the script here; I'll see if it's obvious what I need to do to stop Jenkins building it.

**Comment by jhoblitt** [comment]
@jdswinbank That sounds reasonable to me.  Lets give this a try but leave the `numdiff` installation in `lsstsw/bin/deploy` for the time being so its easy to back out.

Testing is a bit difficult as `lsst_build` requires a working eups installation.  I test by installing `buildbot-scripts` and `lsstsw` then running `lsstswBuild.sh`.

File: README [hunk 1]
```diff
@@ -30,18 +28,23 @@ other parts of the LSST pipeline
...
 
-If you use the "--small" run option then the corresponding comparison file is named:
-detected-sources_small.txt.expected  and the corresponding output directory is called:
-"output_small".
+If you use the "--small" run option then the corresponding output file is
+named detected-sources_small.txt and the corresponding output directory is
+called "output_small".
+
+By default, we use some bespoke Python code to perform the comparison. Specify
+the ``--use-numdiff`` option to ``bin/compare`` to use `Numdiff
```
Comment (jhoblitt) [inline README]: References the now removed `--use-numdiff` flag.

Comment (jdswinbank) [inline README]: Good catch, thanks.

**Pull request merged**